pennylane/labs/resource_estimation/templates/subroutines.py

```diff
@@ -477,6 +477,408 @@ def default_controlled_resource_decomp(
+        """
+        params = {
+            "base_cmpr_op": base_cmpr_op,
+            "num_estimation_wires": num_estimation_wires,
+            "adj_qft_cmpr_op": adj_qft_cmpr_op,
+        }
+        return CompressedResourceOp(cls, params)
+
+    @classmethod
+    def default_resource_decomp(cls, base_cmpr_op, num_estimation_wires, adj_qft_cmpr_op, **kwargs):
+        r"""Returns a dictionary representing the resources of the operator. The
+        keys are the operators and the associated values are the counts.
+
+        Args:
+            base_cmpr_op (CompressedResourceOp): A compressed resource operator, corresponding
+                to the operator that we will be applying controlled powers of.
+            num_estimation_wires (int): the number of wires used for measuring out the phase
+            adj_qft_cmpr_op (CompressedResourceOpor None): A compressed resource operator,
+                corresponding to the adjoint QFT.
+
+        Resources:
+            The resources are obtained from the standard decomposition of QPE as presented
+            in (section 5.2) `Nielsen, M.A. and Chuang, I.L. (2011) Quantum Computation and Quantum
+            Information <https://www.cambridge.org/highereducation/books/quantum-computation-and-quantum-information/01E10196D0A682A6AEFFEA52D53BE9AE#overview>`_.
+        """
+        hadamard = resource_rep(re.ResourceHadamard)
+        ctrl_op = ResourceControlledSequence.resource_rep(base_cmpr_op, num_estimation_wires)
+        if adj_qft_cmpr_op is None:
+            adj_qft_cmpr_op = resource_rep(
+                re.ResourceAdjoint,
+                {
+                    "base_cmpr_op": resource_rep(ResourceQFT, {"num_wires": num_estimation_wires}),
+                },
+            )
+
+        return [
+            GateCount(hadamard, num_estimation_wires),
+            GateCount(ctrl_op),
+            GateCount(adj_qft_cmpr_op),
+        ]
+
+    @staticmethod
+    def tracking_name(base_cmpr_op, num_estimation_wires, adj_qft_cmpr_op) -> str:
+        r"""Returns the tracking name built with the operator's parameters."""
+        base_name = base_cmpr_op.name
+        adj_qft_name = None if adj_qft_cmpr_op is None else adj_qft_cmpr_op.name
+        return f"QPE({base_name}, {num_estimation_wires}, adj_qft={adj_qft_name})"
+
+
+class ResourceIterativeQPE(ResourceOperator):
+    r"""Resource class for Iterative Quantum Phase Estimation (IQPE).
+
+    Args:
+        base_op (~.pennylane.labs.resource_estimation.ResourceOperator): The operator that we
+            will be applying controlled powers of.
+        num_iter (int): the number of mid-circuit measurements made to read out the phase
+        wires (Sequence[int], optional): the wires the operation acts on
+
+    Resources:
+        The resources are obtained following the constuction from `arXiv:quant-ph/0610214v3
+        <https://arxiv.org/pdf/quant-ph/0610214v3>`_.
```

Comment:
Typically better to cite the abs page not the pdf page: https://arxiv.org/abs/quant-ph/0610214v3